modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/DefaultExecutionMediator.java

```diff
@@ -103,17 +109,15 @@ public Object execute(OperationExecutor executor, ExecutionContextAdapter contex
     }
```

Comment:
Be aware that when returning a Mono, finally doesn't work in the same way.  You might need to use `Mono.doAfterTerminate()` instead.